export_options.go

```diff
@@ -74,23 +100,80 @@ func (b XcodebuildTargetInfoProvider) TargetCodeSignEntitlements(target, configu
 	return b.xcodeProj.TargetCodeSignEntitlements(target, configuration)
 }
 
-// projectEntitlementsByBundleID finds the project's main application target, pointed by the provided scheme and configuration,
+func archivableApplicationTarget(xcodeProj *xcodeproj.XcodeProj, scheme *xcscheme.Scheme, configurationName string) (*xcodeproj.Target, error) {
+	archiveEntry, ok := scheme.AppBuildActionEntry()
+	if !ok {
+		return nil, fmt.Errorf("archivable entry not found in project: %s for scheme: %s", xcodeProj.Path, scheme.Name)
+	}
+
+	mainTarget, ok := xcodeProj.Proj.Target(archiveEntry.BuildableReference.BlueprintIdentifier)
+	if !ok {
+		return nil, fmt.Errorf("target not found: %s", archiveEntry.BuildableReference.BlueprintIdentifier)
+	}
+
+	return &mainTarget, nil
+}
+
+func dependentExecutableTargetsOf(applicationtarget xcodeproj.Target) (dependentTargets []xcodeproj.Target) {
+	for _, target := range applicationtarget.DependentExecutableProductTargets(false) {
+		if target.ProductType == appClipProductType {
+			continue
+		}
+
+		dependentTargets = append(dependentTargets, target)
+	}
+	return
+}
+
+// productEntitlementsByBundleID finds the project's main application target, pointed by the provided scheme and configuration,
 // collects it's dependent executable targets and maps each target's entitlements to the target's bundle ID.
-func (g ExportOptionsGenerator) projectEntitlementsByBundleID(xcodeProj *xcodeproj.XcodeProj, scheme *xcscheme.Scheme, configurationName string) (map[string]plistutil.PlistData, error) {
+func (g ExportOptionsGenerator) productEntitlementsByBundleID(xcodeProj *xcodeproj.XcodeProj, scheme *xcscheme.Scheme, configurationName string, appClip bool) (map[string]plistutil.PlistData, error) {
```

Comment:
productEntitlementsByBundleID is not used in GenerateApplicationExportOptions, I suppose you meant it to be used there.

Reply:
this function is a just a leftover code i forgot to remove